test/extended/two_node/tnf_node_replacement.go

```diff
@@ -0,0 +1,1508 @@
+	machineFile := filepath.Join(testConfig.Execution.GlobalBackupDir, testConfig.TargetNode.MachineName+"-machine.yaml")
+
+	// Check if Machine already exists
+	_, err = oc.AsAdmin().Run("get").Args(machineResourceType, testConfig.TargetNode.MachineName, "-n", machineAPINamespace).Output()
+	if err != nil {
+		// Retry Machine creation
+		err = core.RetryWithOptions(func() error {
+			_, err := oc.AsAdmin().Run("create").Args("-f", machineFile).Output()
+			return err
+		}, core.RetryOptions{
+			Timeout:      fiveMinuteTimeout,
+			PollInterval: thirtySecondPollInterval,
+			MaxRetries:   10,
+			ShouldRetry:  services.IsRetryableEtcdError,
+		}, fmt.Sprintf("create Machine %s", testConfig.TargetNode.MachineName))
+
+		if err != nil {
+			return fmt.Errorf("failed to recreate Machine after retries: %v", err)
+		}
+		klog.V(2).Infof("Recreated Machine: %s", testConfig.TargetNode.MachineName)
+	} else {
+		klog.V(2).Infof("Machine %s already exists, skipping recreation", testConfig.TargetNode.MachineName)
+	}
+
+	return nil
+}
+
+// recreateBMCSecret recreates the BMC secret from backup
+func recreateBMCSecret(testConfig *TNFTestConfig, oc *exutil.CLI) error {
+	// Recreate BMC secret with retry
+	bmcSecretFile := filepath.Join(testConfig.Execution.GlobalBackupDir, testConfig.TargetNode.BMCSecretName+".yaml")
+
+	// Check if BMC secret already exists
+	_, err := oc.AsAdmin().Run("get").Args(secretResourceType, testConfig.TargetNode.BMCSecretName, "-n", machineAPINamespace).Output()
+	if err != nil {
+		// Retry BMC secret creation
+		err = core.RetryWithOptions(func() error {
+			_, err := oc.AsAdmin().Run("create").Args("-f", bmcSecretFile).Output()
+			return err
+		}, core.RetryOptions{
+			Timeout:      fiveMinuteTimeout,
+			PollInterval: thirtySecondPollInterval,
+			MaxRetries:   10,
+			ShouldRetry:  services.IsRetryableEtcdError,
+		}, fmt.Sprintf("create BMC secret %s", testConfig.TargetNode.BMCSecretName))
+
+		if err != nil {
+			return fmt.Errorf("failed to recreate BMC secret after retries: %v", err)
+		}
+		klog.V(2).Infof("Recreated BMC secret: %s", testConfig.TargetNode.BMCSecretName)
+	} else {
+		klog.V(2).Infof("BMC secret %s already exists, skipping recreation", testConfig.TargetNode.BMCSecretName)
+	}
+
+	return nil
+}
+
+// ========================================
+// Main Test Functions (in order of execution)
+// ========================================
+
+// backupTargetNodeConfiguration backs up all necessary resources for node replacement
+func backupTargetNodeConfiguration(testConfig *TNFTestConfig, oc *exutil.CLI) string {
+	// Create backup directory
+	var err error
+	backupDir, err := os.MkdirTemp("", backupDirName)
+	o.Expect(err).To(o.BeNil(), "Expected to create backup directory without error")
+
+	// Download backup of BMC secret
+	bmcSecretOutput, err := oc.AsAdmin().Run("get").Args(secretResourceType, testConfig.TargetNode.BMCSecretName, "-n", machineAPINamespace, "-o", yamlOutputFormat).Output()
```

Comment:
I think for these ones I wouldn't worry about using API since the desired result is yaml to file output, avoids extra steps in serializations